cmd/juju/application/deploy.go

```diff
@@ -369,6 +369,33 @@ Constraints can be specified by specifying the '--constraints' option. If the
 application is later scaled out with ` + "`juju add-unit`" + `, provisioned machines
 will use the same constraints (unless changed by ` + "`juju set-constraints`" + `).
 
+Application configuration values can be specified using '--config' option. This
+option accepts either a path to a yaml-formatted file or a key=value pair. 
+Configurtion file provided should be in format
+<charm name>:
+	<option name>: <option value>
+	...
+For example, to deploying 'mediawiki' with the configuration file 'mycfg.yaml'
+that contains:
+
+mediawiki:
+	name: my media wiki
+	admins: me:pwdOne
+	debug: true
+
+use
+
+  juju deploy mediawiki --config mycfg.yaml
+
+To specify key=value pair to set an application option value, use:
+ 
+  juju deploy mediawiki --config name='my media wiki'
+   
+When specifying more than one option value, use:
+
+  juju deploy mediawiki --config name='my media wiki' --config debug=true
```

Comment:
This can also be a mix of --config key=vale --config filename --config filename2   .... etc etc
Last one wins. If a value (incl in file content) in the farthest --config arg takes precedence over an value set earlier.

Reply:
Interesting. I wonder if we want to mention it here on the online docs?

Reply:
that was the intent :-)

Reply:
Great point \o/ addressed...

Reply:
LGTM :octopus:

Reply:
Actually, the help uses this order:

`juju deploy <options> application`

You've put the `--config` after the application.

Reply:
Yes, I did because there can be more things to say about <options> than about 'application'. For one, there can be more then one different option.
This order has been on this help doc for a while and is standard for our commands.